pkg/op/server_http.go

```diff
@@ -378,8 +386,13 @@ func (s *webServer) deviceTokenHandler(w http.ResponseWriter, r *http.Request, c
 	resp.writeOut(w)
 }
 
-func (s *webServer) introspectionHandler(w http.ResponseWriter, r *http.Request, client Client) {
-	if client.AuthMethod() == oidc.AuthMethodNone {
+func (s *webServer) introspectionHandler(w http.ResponseWriter, r *http.Request) {
+	cc, err := s.parseClientCredentials(r)
+	if err != nil {
+		WriteError(w, r, err, s.getLogger(r.Context()))
+		return
+	}
+	if cc.ClientSecret == "" && cc.ClientAssertion == "" {
```

Comment:
isn't that already checked by `parseClientCredentials`?

Reply:
parse checks for `ClientID && ClientAssertion`. It allows clients that only have `ClientID`, in case they are public.

Here we check for `ClientSecret` && `ClientAssertion`, so that we know at least 1 type of secret was provided in the request.

Reply:
ah right